amethyst_network/src/network_socket.rs

```diff
@@ -161,50 +144,56 @@ where
 {
     type SystemData = (WriteStorage<'a, NetConnection<E>>);
 
-    fn setup(&mut self, res: &mut Resources) {
-        Self::SystemData::setup(res);
-    }
-
     fn run(&mut self, mut net_connections: Self::SystemData) {
         for net_connection in (&mut net_connections).join() {
-            let target = net_connection.target;
+            let target = net_connection.target_receiver;
 
             if net_connection.state == ConnectionState::Connected
                 || net_connection.state == ConnectionState::Connecting
             {
-                self.tx
+                self.transport_sender
                     .send(InternalSocketEvent::SendEvents {
                         target,
                         events: net_connection.send_buffer_early_read().cloned().collect(),
                     })
                     .expect("Unreachable: Channel will be alive until a stop event is sent");
             } else if net_connection.state == ConnectionState::Disconnected {
-                self.tx
+                self.transport_sender
                     .send(InternalSocketEvent::Stop)
                     .expect("Already sent a stop event to the channel");
             }
         }
 
-        for raw_event in self.rx.try_iter() {
+        for (counter, raw_event) in self.transport_receiver.try_iter().enumerate() {
```

Comment:
👍 good solution I didn't think of it this way and this is def simpler! Good stuff!